File: B13/Array/Hemanth/Two_sum.java
//Leetcode Link : https://leetcode.com/problems/two-sum/

//Using Hashmaps

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] a = new int[2];                        // Array used to store pair of indeces.
        Map<Integer,Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(target-nums[i])){
                a[1]=i;
                a[0]=map.get(target-nums[i]);
            }else{
            map.put(nums[i],i);
            }
        }
        return a;
    }
}

//Time Complexity : O(n)
//Space Complexity : O(n)

File: B13/Subhendu/Array/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ## https://leetcode.com/problems/two-sum/

        used_num = dict()

        for i in range(len(nums)):
            if (target - nums[i]) in used_num:
                return [used_num[target - nums[i]], i]
            used_num[nums[i]] = i